### PR TITLE
Add handling of movq instruction with XMM resigters.

### PR DESCRIPTION
**Pull request by gartung**
fixes https://github.com/igprof/igprof/issues/93

### CODE REVIEW

File: src/hook.cc [hunk 1]
```diff
@@ -991,7 +991,9 @@ parse(const char *func, void *address, unsigned *patches)
     //ENDBR64
     else if (insns[0] == 0xf3 && insns[1] == 0x0f && insns[2] == 0x1e && insns[3] == 0xfa )
       n +=4, insns += 4;
-
+    //MOVQ with XMM registers.
+    else if (insns[0] == 0x66 && insns[1] == 0x48 && insns[2] == 0xf)
+      n +=2, insns +=2;
```
[the remark this answers is not in the document]
Comment (ktf) [reply src/hook.cc:996]: i mean insns += 3

[the remark this answers is not in the document]
Comment (ktf) [reply src/hook.cc:996]: Actually, looking at the disassembly you have in #93, I would have expected n+=5 insns+=5 and a copy for the last two bytes, no?

[the remark this answers is not in the document]
Comment (gartung) [reply src/hook.cc:996]: This is still a work in progress. I am waiting for build and test with CMSSW. I didn't see a way to submit a draft pull request.

[the remark this answers is not in the document]
Comment (gartung) [reply src/hook.cc:996]: You are correct. I will need to skip more. I will test your suggestion.
With this change I get a different warning
```
*** IgProf(26542, 1669997554.893): malloc (0x7f71427107b0): instrumenting 7 bytes into 0x7f714401f004
*** IgProf(26542, 1669997554.893): calloc (0x7f7142711550) + 0x4: unrecognised prologue (found 0xf 0x6e 0xc7 0x66)
*** IgProf(26542, 1669997554.895): realloc (0x7f714270e2b0): instrumenting 6 bytes into 0x7f7144020004
*** IgProf(26542, 1669997554.897): posix_memalign (0x7f7142710880): instrumenting 6 bytes into 0x7f7144021004
*** IgProf(26542, 1669997554.897): memalign (0x7f714270f260) + 0x4: unrecognised prologue (found 0xf 0x6e 0xc7 0x66)
```

[the remark this answers is not in the document]
Comment (gartung) [reply src/hook.cc]: With the change you suggested I get 
```
*** IgProf(10499, 1670009958.458): memory profiler: reporting memory use without allocation overhead
*** IgProf(10499, 1670009958.460): malloc (0x7f852b93a7b0): instrumenting 7 bytes into 0x7f852d249004
*** IgProf(10499, 1670009958.461): calloc (0x7f852b93b550): instrumenting 7 bytes into 0x7f852d24a004
*** IgProf(10499, 1670009958.463): realloc (0x7f852b9382b0): instrumenting 6 bytes into 0x7f852d24b004
*** IgProf(10499, 1670009958.464): posix_memalign (0x7f852b93a880): instrumenting 6 bytes into 0x7f852d24c004
*** IgProf(10499, 1670009958.466): memalign (0x7f852b939260): instrumenting 7 bytes into 0x7f852d24d004
```

File: src/hook.cc [hunk 2]
```diff
@@ -991,7 +991,9 @@ parse(const char *func, void *address, unsigned *patches)
     //ENDBR64
     else if (insns[0] == 0xf3 && insns[1] == 0x0f && insns[2] == 0x1e && insns[3] == 0xfa )
       n +=4, insns += 4;
-
+    //MOVQ with XMM registers.
+    else if (insns[0] == 0x66 && insns[1] == 0x48 && insns[2] == 0xf)
```
Comment (ktf) [inline src/hook.cc:995]: you probably should copy the remaining bytes in insns[3] and insns[4]

Comment (gartung) [reply src/hook.cc:995]: The only mention I see is
https://github.com/igprof/igprof/blob/58942b67bc2b35a5c8e38024e322b8d3d28800c8/src/hook.cc#L1177

Comment (ktf) [reply src/hook.cc:995]: indeed maybe it's not needed.

**Pull request closed without merging**